### ISSUE TITLE
Inventory layout

### ISSUE THREAD

**Issue by AKryukov92** [DreamBlocks/DreamBlocks#38]
Crafting guide is needed to let people know what recipes exist in the game.

**Comment by LordStephen77** [comment]
it's true. I did not do it because the recipes are the same as Minetest and Minecraft. But it's a good idea.

**Comment by AKryukov92** [comment]
Even if we are going to stick with MC and MT recipes, we need to show that recipes we have _not_ implemented yet.
For example there is no furnace at this moment.

I'm asking @tobyplowy to draw layout of crafting guide.

**Comment by LordStephen77** [comment]
@AKryukov92 Toby is our only hope! I work on the website, I've improved it a lot. I can not even deal with graphic content.

@tobyplowy Toby takes away all your creativity :)

**Comment by tobyplowy** [comment]
@LordStephen77 dount fear Toby is here! No but really thanks for the kind words it means alot!

@AKryukov92 yes will do :)

**Comment by tobyplowy** [comment]
![screenshot-_33_ - kopie](https://user-images.githubusercontent.com/16853304/27597350-5d6d4b16-5b62-11e7-9513-9f72e0018c6d.png)

**Comment by tobyplowy** [comment]
![screenshot-_33_ - kopie](https://user-images.githubusercontent.com/16853304/27597657-485e9aa8-5b63-11e7-96c3-16fd7823344d.png)

**Comment by AKryukov92** [comment]
In this layout it is unclear how recipe should be drawn.

**Comment by tobyplowy** [comment]
@AKryukov92 let me explain my idea
all the slots shuld have a item or Block in them if you click on that item or Block it will bring you back to the craft grid (inventory or crafting table) and it shows you a transparant overlay on the craft grid of the selected crafting recipe you just put the correct stuff in and haza it's done 

Is this a good idea or shuld I redesign?

**Comment by AKryukov92** [comment]
I'v got your idea. But why transparent overlay?
When player clicks on item or block, it is possible to open inventory, check backpack for resources and place them to crafting grid.

**Comment by AKryukov92** [comment]
By the way, I would like to know which recipes I can craft at this moment.
There should be a way to tell player: "you can craft this recipe now".

**Comment by tobyplowy** [comment]
> But why transparent overlay?

Because if it's not transparent People May think it's already in the Slots and they will get frustrated and confused

**Comment by tobyplowy** [comment]
> "you can craft this recipe now"

Yes maybe make the items in the craft guide that you can't craft (Because you don't have the Resources ) have a Red outline or overlay?

**Comment by AKryukov92** [comment]
1. When user fills crafting grid with transparent overlay, there should be a simple way to remove this overlay. Will you make additional button in the inventory?
2. Red outline will work good if you just say player "you can't craft this". But it doesn't explain the reason.
Player will need to click this recipe, look to overlay and then check "what else recources I need?".
3. How to display furnace recipes?

**Comment by tobyplowy** [comment]
![nassssamloos](https://user-images.githubusercontent.com/16853304/27600578-b87fc31c-5b6c-11e7-9361-8744ed8e6118.png)

**Comment by AKryukov92** [comment]
Please answer to my previous message.

**Comment by tobyplowy** [comment]
> When user fills crafting grid with transparent overlay, there should be a simple way to remove this overlay. Will you make additional button in the inventory?

Yes I will add that

> Red outline will work good if you just say player "you can't craft this". But it doesn't explain the reason.
> Player will need to click this recipe, look to overlay and then check "what else recources I need?".

If You Hover over an item with Red outline it should show a tool tip that says the reason like you don't have the correct items for this or you need a furnace or crafting table for this or you need a hier experience level To make this

> How to display furnace recipes?

You need to be in the furnace otherwise It will not work If You click on the recipe (if you are in a crafting table and you try this it should say you need a furnace for this)

**Comment by AKryukov92** [comment]
What about of displaying all required resources in crafting guide?
In this case you could mark missing resources by the red outline. No tooltip, no transparent overlays required.

What is the reasoning behind current layout?

**Comment by AKryukov92** [comment]
Sorry. I missed the button on my phone.

**Comment by tobyplowy** [comment]
> What about of displaying all required resources in crafting guide?

I don't really understand That question
Are you asking me to add a crafting grid to the crafting guide to show required resources?
(That shuld happen in the crafting table)

**Comment by tobyplowy** [comment]
Yes that Command and close butten is evil! :D

**Comment by AKryukov92** [comment]
> Are you asking me to add a crafting grid to the crafting guide to show required resources?

Yes. I think we should see different options before we choose any.

**Comment by tobyplowy** [comment]
@AKryukov92 ok but this reduses space on the craft guide

**Comment by tobyplowy** [comment]
![screenshot-_33_ - kopie](https://user-images.githubusercontent.com/16853304/27606214-b441d50c-5b7f-11e7-8bb1-b72dc1fb8269.png)

**Comment by tobyplowy** [comment]
NOTE the crafting grid shuld cange depending on what you select

**Comment by tobyplowy** [comment]
i moved caft guide button i added a trash slot this dhuld remove whats inside 
and i added a clear craft overlay button
![screenshot-_33_](https://user-images.githubusercontent.com/16853304/27606428-82578392-5b80-11e7-9f0f-c49b0329d496.png)

**Comment by AKryukov92** [comment]
Dont forget that crafting grid can be 3x3 in workbench. It should not look too different.

**Comment by tobyplowy** [comment]
Yes thats why I made it the way I did in the first picture

**Comment by tobyplowy** [comment]
but it works with a 3x3
![screenshot-_33_](https://user-images.githubusercontent.com/16853304/27606720-a7a8a968-5b81-11e7-8d53-1012ca94c72f.png)

**Comment by AKryukov92** [comment]
Will selected item (result of the recipe) be highligted or be drawn near grid?
Advantages of second layout of crafing guide
* It is possible to make button "place resources to table". I'm not sure if this needs to be implemented. What is the purpose of crafting grid if we can skip placing items there?
* It is easier to show which resources player needs to create an item.
* If you are interested in resources of multiple items, you don't need to switch to inventory all the time.

Disadvantages:
* less items in one page of the guide
* if there will be no "place resources to grid" button, then player needs to memorize recipe. This is probably a good thing. After some time player will not need crafting guide at all.

**Comment by tobyplowy** [comment]
I think it's best to have it highligted

**Comment by tobyplowy** [comment]
Will you program all this code?

**Comment by AKryukov92** [comment]
Probably. I'm not sure if it will be soon. There are some other inventory issues blocking this work.

**Comment by AKryukov92** [comment]
How "search for item" should behave? What text player should type there?

**Comment by tobyplowy** [comment]
@AKryukov92 well for example if you type pickaxe you see all crafteble pickaxes or if you type diamond pickaxe you only find a diamond pickaxe 

You can also lookup stick and you will find stick 

Its the same like in Minetest creative menu (item search)

**Comment by tobyplowy** [comment]
You should be able to type and find everything

**Comment by tobyplowy** [comment]
> if there will be no "place resources to grid" button, then player needs to memorize recipe. This is probably a good thing. After some time player will not need crafting guide at all.

Yes that is a good thing

Bdw I never use a craft guide because I don't need it but some people do so we need to add it :)

**Comment by tobyplowy** [comment]
Also what do you think of my gui (Graphic user interface) style

**Comment by Gerold55** [comment]
Would it be possible 2 make it so the inventory grid has an outline so you can distinguish between crafting grid and inventory grid? Or make it so that there's a little bit more room in between them and also could there be an option of autocrafting? I could come up with a concept design in a little bit.

**Comment by Gerold55** [comment]
Could also make crafting guide an option so add a button to turn it on and off in the menu b4 u start a world and in the options menu when in the world

**Comment by AKryukov92** [comment]
Both things are possible. But ,first of all they should be drawn in a picture.
At this moment, there are no "options" dialog. As long as we have only one option, it is better to place it in the dialog where player chooses world size.

**Comment by Gerold55** [comment]
Well 4 options u could add volume rocker and gui size rocker

**Comment by Gerold55** [comment]
@tobyplowy shouldn't the button 4 the craft guide be in the same position when in inventory also?

**Comment by Gerold55** [comment]
![menu](https://user-images.githubusercontent.com/13843095/27769758-2cfec1a8-5eff-11e7-9e89-36c07909d5a7.png)

**Comment by tobyplowy** [comment]
> shouldn't the button 4 the craft guide be in the same position when in inventory also?

That may be hard to achieve (it will probably reduce space on the GUI)

**Comment by tobyplowy** [comment]
@Gerold55 Nice menu!

Now just the pixel text and it's perfect also maybe remove the hot bar

**Comment by Gerold55** [comment]
Yeah I know it's just a proof of concept.

**Comment by Gerold55** [comment]
@tobyplowy is it possible to make it so it's a bit easier 2 distinguish between the crafting grid and the inventory space?

**Comment by tobyplowy** [comment]
> is it possible to make it so it's a bit easier 2 distinguish between the crafting grid and the inventory space?

I will look into that later

> Yeah I know it's just a proof of concept.

Will you program that?

**Comment by Gerold55** [comment]
I don't know how to code in Java I'm more of an artist and an imager I can come up with some ideas and run it past u and do concept art 4 said ideas

**Comment by LordStephen77** [comment]
Achievements?!?

**Comment by Gerold55** [comment]
Just a suggestion eventually tho something similar 2 minecraft

**Comment by AKryukov92** [comment]
Guys. Please keep your proposed visual layouts closer to actual state of the game.

I agree that crafting grid in inventory layout should be separated from backpack.

**Comment by AKryukov92** [comment]
@LordStephen77 what do you think about Munro font? It is free and @tobyplowy liked it.
https://www.fontsquirrel.com/fonts/Munro

**Comment by Gerold55** [comment]
Would it b possible to create "our" own font?

**Comment by Gerold55** [comment]
@LordStephen77 we know that it is way ahead but you need to plan ahead sometimes and to get features planned and ready 2 b implemented

**Comment by AKryukov92** [comment]
Why would you want to create your own font?
I am agree, that we should plan ahead. But making single button "achievements" is not a planning.
When you make detailed description of achievements, visual layout of achievements screen and all associated art, then it will make sense to add a button to menu.

**Comment by Gerold55** [comment]
I added the button because it's a concept art I'll work on some more concept art soon for the achievements

**Comment by AKryukov92** [comment]
Please add to concept of the button images of "hover" state (when you point it with mouse)

**Comment by Gerold55** [comment]
It should be the same as the menu because I used those buttons

**Comment by Gerold55** [comment]
![achievements](https://user-images.githubusercontent.com/13843095/27789651-5d6855a4-5fbb-11e7-8849-d7b925bf32d8.png)

**Comment by AKryukov92** [comment]
You need to describe your concepts. What is it?

**Comment by Gerold55** [comment]
@AKryukov92 oops sry bout that. That's the achievements panel needs some work done still but was hoping @tobyplowy could tweak it so it matched the theme of the other inventories

**Comment by tobyplowy** [comment]
@Gerold55 yup will do later

**Comment by tobyplowy** [comment]
@AKryukov92 @LordStephen77 is it worth my time to make the GUI for achievements

**Comment by Gerold55** [comment]
@tobyplowy 2 me eventually it could be a feature a concept is good for eventually because right now a simple achievements could be could
